optimizely/optimizely.py

```diff
@@ -203,8 +203,10 @@ def _get_feature_variable_for_type(self, feature_key, variable_key, variable_typ
     if not variable:
       return None
 
-    # Return None if type differs
-    if variable.type != variable_type:
+    # For non-typed method, use type of variable; else, return None if type differs
+    if not variable_type:
+      variable_type = variable.type
```

Comment:
You can replace the 2 lines with:
```
variable_type = variable_tupe or variable.type
```

Reply:
Replaced the 2 lines with your suggestion. Neat trick.